core/dbt/cli/flags.py

```diff
@@ -239,6 +241,32 @@ def _assign_params(
             version_check = getattr(self, "VERSION_CHECK", True)
             object.__setattr__(self, "LOG_PATH", default_log_path(project_dir, version_check))
 
+        # Re-apply the original vars when performing a retry
+        if getattr(self, "WHICH", "") == "retry":
+            previous_state = PreviousState(
+                state_path=Path(
+                    ctx.params.get("state", "") or getattr(self, "TARGET_PATH") or "."
+                ),
+                target_path=Path(
+                    getattr(self, "TARGET_PATH")
+                    or os.path.join(ctx.params["project_dir"], "target")
+                ),
+                project_root=Path(ctx.params.get("project_dir") or "."),
+            )
+
+            prev_vars = (
+                previous_state.results.args.get("vars", {}) if previous_state.results else None
+            )
+
+            if prev_vars:
+                curr_vars = getattr(self, "VARS", {})
+                object.__setattr__(self, "VARS", {**prev_vars, **curr_vars})
```

Comment:
Looks like you choose to do the combine and later one win!
Note we do have this dict [here](https://github.com/dbt-labs/dbt-core/blob/3f7f7de1791ee54bfe313ca033aa1b4046377900/core/dbt/task/retry.py#L22)

Reply:
Let's check with @graciegoheen. What do you think, Grace?

Reply:
The upside of combining is that if you only want to overwrite one var, you don't need to supply the all the vars.
The downside of combining is that if you want to stop supplying a var, you will have to set it to None or the default value of it if it is already defined since user supplied takes highest presidency.

Reply:
My personal preference is the second because if a person is already supplying vars again, making it more explicit is better.